src/Simulation/Intrinsic/Convert/Convert.qs

```diff
@@ -31,6 +31,17 @@ namespace Microsoft.Quantum.Convert {
     }
     
     
+    /// # Summary
+    /// Converts a given big integer to an equivalent integer, if possible.
+    /// The function returns a pair of the resulting integer and a Boolean flag
+    /// which is true, if and only if the conversion was possible.
+    /// # Remarks
+    /// See [C# BigInteger constructor](https://docs.microsoft.com/dotnet/api/system.numerics.biginteger.-ctor?view=netframework-4.7.2#System_Numerics_BigInteger__ctor_System_Int64_) for more details.
+    function BigIntAsInt(a : BigInt) : (Int, Bool) {
```

Comment:
I really like this as a signature, as per our discussion on the other thread. I think that this signature is distinct enough from other `XAsY` functions, however, that we should call that out in the name. Perhaps `MaybeBigIntAsInt`, thinking of `(Int, Bool)` as a kind of "Int option" or "Int?" pattern? If so, I'm happy to add in M.Q.Convert new functions `Definitely<'T>(maybeValue : ('T, Bool)) : 'T` and `BigIntAsInt(a : BigInt) : Int` that `fail` if the conversion fails.